fe/fe-core/src/main/java/com/starrocks/system/BackendCoreStat.java

```diff
@@ -14,7 +14,8 @@ public class BackendCoreStat {
     private static AtomicInteger cachedAvgNumOfHardwareCores = new AtomicInteger(-1);
 
     public static void setNumOfHardwareCoresOfBe(long be, int numOfCores) {
-        if (numOfHardwareCoresPerBe.putIfAbsent(be, numOfCores) == null) {
+        Integer previous = numOfHardwareCoresPerBe.put(be, numOfCores);
```

Comment:
previous == null || previous != numOfCores can merged into one stmt
modify it to if (numOfHardwareCoresPerBe.putIfAbsent(be, numOfCores)!=numOfCores)